plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -970,6 +970,10 @@ public void testDate()
         checkIsDoubled(someZone, dateOfLocalTimeChangeBackwardAtMidnightInSomeZone.atStartOfDay().minusMinutes(1));
 
         DataTypeTest testCases = DataTypeTest.create(true)
+                .addRoundTrip(dateDataType(), LocalDate.of(1582, 10, 4)) // before julian->gregorian switch
+                // PostgreSQL plus 10 days when the date is in the range of 1582 Oct 5 and 14
+                // .addRoundTrip(dateDataType(), LocalDate.of(1582, 10, 5))
```

Comment:
That might be effect of using `dateWriteFunctionUsingLocalDate`.

try write function with 
```
PGobject pgObject = new PGobject();
pgObject.setType("date");
pgObject.setValue(localDate.toString());
stmt.setObject(index, pgObject);
```

Reply:
The issue happened when reading result set. Going to change to use `ResultSet.getString()` likes SQL Server connector. 

<img width="729" alt="Screen Shot 2021-11-30 at 9 50 20" src="https://user-images.githubusercontent.com/6237050/143966149-fa4f4200-cfaa-4b08-980b-23034fa24878.png">

Reply:
By the way, PostgreSQL allows `infinity` and `-infinity` in date type column. Do you have any recommendation to handle those values? I will file and issue and leave TODO comment if needed.

Reply:
I don't think we can handle them in any reasonable manner. We should fail the read (throw).

Reply:
> Going to change to use ResultSet.getString() likes SQL Server connector.

👍